app/lib/mammoth/status_origin.rb

```diff
@@ -50,13 +50,19 @@ def bulk_add_mammoth_pick(statuses, user)
         bulk_reasons(reasons)
     end 
     
-    def bulk_reasons(reasons)
+    def bulk_reasons(user, reasons)
+        user_list_key = key(user[:acct])
+        Rails.logger.debug "USER LIST KEY #{user_list_key}"
         redis.pipelined do |p|
             reasons.each do |r|
+                p.zadd(user_list_key, 0, r[:key])
                 p.zadd(r[:key], r[:id], r[:reason])
```

Comment:
add status key as an entry in the user list key, so we'll have an index of all the keys we need to purge.